src/Pos/Launcher/Options.hs

```diff
@@ -14,14 +14,36 @@ type DistrOption = Maybe (Int, Int)
 
 panicConflicting :: panic
 panicConflicting =
-    error $ "Conflicting distribution options were enabled. " <>
-            "Choose one at most or nothing."
+    error "Conflicting distribution options were enabled. \
+          \Choose one at most or nothing."
 
-stakesDistr :: DistrOption -> DistrOption -> Bool -> StakeDistribution
-stakesDistr Nothing Nothing False = def
-stakesDistr (Just (nodes, coins)) Nothing False =
+stakesDistr
+    :: DistrOption
+    -> DistrOption
+    -> Maybe (Int, Int, Integer, Double)
+    -> Bool
+    -> StakeDistribution
+stakesDistr Nothing Nothing Nothing False = def
+stakesDistr (Just (nodes, coins)) Nothing Nothing False =
     FlatStakes (fromIntegral nodes) (mkCoin (fromIntegral coins))
-stakesDistr Nothing (Just (nodes, coins)) False =
+stakesDistr Nothing (Just (nodes, coins)) Nothing False =
     BitcoinStakes (fromIntegral nodes) (mkCoin (fromIntegral coins))
-stakesDistr Nothing Nothing True = ExponentialStakes
-stakesDistr _ _ _ = panicConflicting
+stakesDistr Nothing Nothing (Just (richs, poors, coins, richShare)) False =
+    checkConsistency $ RichPoorStakes {..}
+  where
+    sdRichmen = fromIntegral richs
+    sdPoor = fromIntegral poors
+
+    totalRichStake = round $ richShare * fromIntegral coins
+    totalPoorStake = coins - totalRichStake
+    richStake = totalRichStake `div` fromIntegral richs
+    poorStake = totalPoorStake `div` fromIntegral poors
```

Comment:
We have the same situation with testnet stakes. I think it's not really important.